fleet_adapter_mir/MiRCommandHandle.py

```diff
@@ -0,0 +1,986 @@
+                and self.robot_state.mode.mode != RobotMode.MODE_IDLE):
+            self.node.get_logger().info(
+                '[ABORT] {self.name}: Robot stop called!'
+            )
+
+    def follow_new_path(self,
+                        waypoints,
+                        next_arrival_estimator,  # function!
+                        path_finished_callback):
+        waypoint_indices = ' '.join([str(x.graph_index) for x in waypoints])
+        print(f"Following new Path: {waypoint_indices}")
+
+        self.stop()
+        self.current_task_id += 1
+
+        self.rmf_path_requested = True
+
+        # Obtain plan waypoints ===============================================
+        waypoints = copy.copy(waypoints)
+
+        self.rmf_remaining_path_waypoints = [
+            (i, waypoints[i]) for i in range(len(waypoints))
+        ]
+
+        # We reverse this list so that we can pop it instead of traversing
+        # it using an index (which is more Pythonic)
+        self.rmf_remaining_path_waypoints.reverse()
+
+        # Construct robot state path list =====================================
+        self.rmf_robot_state_path_locations = []
+
+        for (_, waypoint) in self.rmf_remaining_path_waypoints:
+            # Split timestamp into decimal and whole second portions
+            _sub_seconds, _seconds = math.modf(waypoint.time.timestamp())
+
+            _msg = Location()
+            _msg.x, _msg.y, _msg.yaw = waypoint.position
+            _msg.t.sec, _msg.t.nanosec = int(_seconds), int(_sub_seconds * 1e9)
+
+            self.rmf_robot_state_path_locations.append(_msg)
+
+        if not self.dry_run:
+            state_id = MiRState.READY
+            self.mir_api.status_put(state_id)
+
+        def path_following_closure():
+            _current_waypoint = None
+            _next_waypoint = None
+
+            # LOOP ============================================================
+            # Kept alive if paused
+            while ((self.rmf_remaining_path_waypoints or self.paused)
+                    or _current_waypoint):
+
+                # DEBUG PRINTOUT
+                # print([str(x[1].graph_index) for x in self.rmf_remaining_path_waypoints])
+
+                if not self.paused:  # Skipped if paused
+                    if _current_waypoint is None:
+                        _, _current_waypoint = (
+                            self.rmf_remaining_path_waypoints.pop()
+                        )
+                        self.rmf_path_requested = True
+
+                    waypoint_leave_msg = _current_waypoint.time
+                    #ros_waypoint_leave_time = (
+                    #    waypoint_leave_msg.sec
+                    #    + waypoint_leave_msg.nanosec / 1e9
+                    #)
+                    ros_waypoint_leave_time = waypoint_leave_msg
+
+                    ros_now = self.node.get_clock().now().nanoseconds / 1e9
+                    next_mission_wait = (
```

Comment:
That is good to know! Yeah I think it would be a good idea to keep it as an open issue, the changes required should be a low hanging fruit when we get to perform tests again in the future.

Opened issue https://github.com/osrf/fleet_adapter_mir/issues/18